### DISCUSSION TITLE
I define a custom meta type for columns, but VsCode is not inferring type

### DISCUSSION THREAD

**Discussion by cheesyMan** [TanStack/table#4072] [Q&A]
Hi everyone,
I'm pretty new to React-Table (v8) and TS, and I'm having some troubles with column meta-type.

As explained in official docs, I create a table:
`const table = createTable().setRowType<MyCustomDataType>();`

then, I define a **custom meta-type** for my columns to set CSS classes for both headers and cells, and I set this meta-type for the table with `setColumnMetaType()` method:
```
type MetaType = {
    headerClass?: string;
    cellClass?: string;
};

table.setColumnMetaType<MetaType>();
```

**BUT**, when I start to define my columns, **VSCode** makes **no type inference** on the meta property:

![Snipaste_2022-06-29_15-52-43](https://user-images.githubusercontent.com/55018265/176453727-be8776f5-24ef-47ef-b9f5-b25faca5b886.jpg)

![Snipaste_2022-06-29_15-49-02](https://user-images.githubusercontent.com/55018265/176453768-37127409-2451-4546-bbae-a1440538a45d.jpg)

Any hints? Thanks :)

**Comment by AzzouQ** [comment]
As stated in the doc here [here](https://tanstack.com/table/v8/docs/api/core/column-def#meta)
> ⚠️ Due to generic limitations, This meta object must be typecast to the correct type before usage.

What you need to do is casting your `meta` prop like
```jsx
(columnDef.meta as MetaType).headerClass...
```
